dbms/src/Common/ThreadedWorker.h

```diff
@@ -204,8 +204,8 @@ class ThreadedWorker
             auto error = getCurrentExceptionMessage(false);
             LOG_ERROR(log, "{}#{} meet error: {}", getName(), thread_idx, error);
             auto cancel_reason = fmt::format("{} failed: {}", getName(), error);
-            result_queue->cancelWith(cancel_reason);
             source_queue->cancelWith(cancel_reason);
+            result_queue->cancelWith(cancel_reason);
```

Comment:
![42d6a878-4885-40bb-a647-2d2c20fd8a9c](https://github.com/pingcap/tiflash/assets/7493273/290df2c0-892c-4bc5-812a-3dc507b21f16)
Change order of cancel to fix above unittest. Check [code](https://github.com/pingcap/tiflash/blob/master/dbms/src/Common/tests/gtest_threaded_worker.cpp#L309)

Reply:
/cc @breezewish